src/main/java/io/confluent/idesidecar/restapi/credentials/Credentials.java

```diff
@@ -0,0 +1,120 @@
+package io.confluent.idesidecar.restapi.credentials;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonSubTypes.Type;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import io.confluent.idesidecar.restapi.exceptions.Failure;
+import io.confluent.idesidecar.restapi.models.ConnectionSpec.SecurityProtocol;
+import io.confluent.idesidecar.restapi.models.ConnectionSpec.SslIdentificationAlgorithm;
+import io.quarkus.runtime.annotations.RegisterForReflection;
+import io.soabase.recordbuilder.core.RecordBuilder;
+import io.vertx.core.MultiMap;
+import jakarta.validation.constraints.NotNull;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Base interface for credentials objects used with Kafka and Schema Registry clients.
+ */
+@JsonTypeInfo(use = JsonTypeInfo.Id.DEDUCTION)
+@JsonSubTypes({
+    @Type(value = BasicCredentials.class),
+    @Type(value = ApiKeyAndSecret.class)
+})
```

Comment:
To achieve the `oneOf` pattern as I recommended in my other comment, we'll have to add the`@Schema` annotation here (so that it doesn't need to be repeated everywhere `Credentials` is declared as a field):

```suggestion
@JsonTypeInfo(use = JsonTypeInfo.Id.DEDUCTION)
@JsonSubTypes({
    @Type(value = BasicCredentials.class),
    @Type(value = ApiKeyAndSecret.class)
})
@Schema(
    description =
        "The credentials for the Kafka cluster, or null if not authentication is required",
    oneOf = {BasicCredentials.class, ApiKeyAndSecret.class},
    nullable = true
)
```

I tried this locally and it generates the `Credentials` schema as follows:

```json
      "Credentials" : {
        "description" : "The credentials for the Kafka cluster, or null if not authentication is required",
        "type" : "object",
        "oneOf" : [ {
          "$ref" : "#/components/schemas/BasicCredentials"
        }, {
          "$ref" : "#/components/schemas/ApiKeyAndSecret"
        } ],
        "nullable" : true
      },
```

Reply:
Great suggestion. I'll try to add this on the `ConnectionSpec.KafkaCluster.credentials` and `ConnectionSpec.SchemaRegistry.credentials` fields. 

This generates in the `openapi.yaml` file
```yaml
    SchemaRegistryConfig:
      ...
      properties:
        ...
        credentials:
          description: "The credentials for the Schema Registry, or null if no authentication is required"
          type: object
          allOf:
          - $ref: "#/components/schemas/Credentials"
          oneOf:
          - $ref: "#/components/schemas/BasicCredentials"
          - $ref: "#/components/schemas/ApiKeyAndSecret"
          nullable: true
```
and
```yaml
    KafkaClusterConfig:
      ...
      properties:
        ...
        credentials:
          description: "The credentials for the Kafka cluster, or null if no authentication is required"
          type: object
          allOf:
          - $ref: "#/components/schemas/Credentials"
          oneOf:
          - $ref: "#/components/schemas/BasicCredentials"
          - $ref: "#/components/schemas/ApiKeyAndSecret"
          nullable: true
```
I think this is better for a few reasons:

1. The description can be more specific. 
2. It allows other uses of `Credentials` to also be customized a bit more.
3. This allows us to put a different subset of `Credential` implementations each place where it's used, if not all `Credential` subtypes apply. For example, SR clients can inherit the credentials used for connecting to the Kafka cluster, and we'd probably implement that with an `InheritedCredentials` that we would _not_ want to use on the Kafka cluster.